### ISSUE TITLE
Web Page Widget not refreshing to display new content.

### ISSUE THREAD

**Issue by btfrise** [Rise-Vision/widget-web-page#84]
**How to recreate**

1. Create a sample web page with an image tab and upload it to the web. [Example](https://storage.googleapis.com/risemedialibrary-e113f8d6-de1c-4c26-b160-4ef1c4589957/example.html)get,
2. Create a new Presentation with a Web Page Widget, set it's refresh to 1 or 5 minutes, and point it at the above URL.
3. Assign to a Display.
4. Either upload a new image with the same filename to the same folder (E.g. new image at same URL the .html page references), or change the url in the .html page to a new image.

**What happens**
Web Page Widget on the Display doesn't update to the new image, even after a restart.

**What should happen**
The web page should reload per the refresh set on the widget, and the new image should appear.

**Comment by btfrise** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [(1) Issue 84 When Web Page is updated, Web Page Widget doesn't refresh with new content.](https://trello.com/c/3laQN0b3/8139-1-issue-84-when-web-page-is-updated-web-page-widget-doesnt-refresh-with-new-content)

**Comment by sheadarlison** [comment]
@btfrise I couldn't recreate this using my hosting. I followed the steps and it updated on the refresh.

@stulees could this be happening because of how our/their headers are set? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control

**Comment by sheadarlison** [comment]
@btfrise I think that's the issue, see the difference below, but Stu will need to confirm.

Website: https://www.giftofspeed.com/cache-checker/
Your URL: https://storage.googleapis.com/risemedialibrary-e113f8d6-de1c-4c26-b160-4ef1c4589957/example.html
My URL: http://sheadarlison.com/test.html

![screen shot 2018-06-13 at 10 56 03 am](https://user-images.githubusercontent.com/9273546/41359407-6fa5c768-6ef8-11e8-9acd-ce71857c53a9.png)

![screen shot 2018-06-13 at 10 56 35 am](https://user-images.githubusercontent.com/9273546/41359421-776e0eb0-6ef8-11e8-9ae0-8ed49580a44a.png)

**Comment by stulees** [comment]
@sheadarlison @btfrise  Yes, Storage responds with a `cache-control: max-age=86400` header (24 hours) for both the HTML and image. Since the code in this HTML does not leverage using `rise-storage` web component (or in the future the common library for working with RLS) for retrieving the storage image file, this webpage will adhere to the cache-control header for all files. 

Webpage Widget can not do anything about this.

**Comment by sheadarlison** [comment]
@btfrise given the above is it okay if we close this? Do you need any more info from us? 

Thanks

**Comment by btfrise** [comment]
@stulees @sheadarlison I know it's outside of the scope of our support, but could you lead me in the right direction to help our user setup his local webserver with the correct header to update this image?

**Comment by stulees** [comment]
@btfrise You can advise the user to configure the cache-control and/or expires response headers to prevent caching. For example:

```
Cache-Control: no-cache, no-store, must-revalidate
```
Documentation - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
